logging/logging.go

```diff
@@ -0,0 +1,93 @@
+package logging
+
+import (
+	"fmt"
+	"io"
+	"strings"
+	"sync"
+
+	"github.com/charmbracelet/log"
+)
+
+type Logger struct {
+	enableInfo        bool
+	enableTracing     bool
+	mutraceSubsystems sync.Mutex
+	traceSubsystems   map[string]bool
+	stdoutLogger      *log.Logger
+	stderrLogger      *log.Logger
+	infoLogger        *log.Logger
+	warnLogger        *log.Logger
+	debugLogger       *log.Logger
+	traceLogger       *log.Logger
+}
+
+func NewLogger(stdout io.Writer, stderr io.Writer) *Logger {
+	return &Logger{
+		enableInfo:      false,
+		enableTracing:   false,
+		stdoutLogger:    log.NewWithOptions(stdout, log.Options{}),
+		stderrLogger:    log.NewWithOptions(stderr, log.Options{}),
+		infoLogger:      log.NewWithOptions(stdout, log.Options{Prefix: "info"}),
+		warnLogger:      log.NewWithOptions(stderr, log.Options{Prefix: "warn"}),
+		debugLogger:     log.NewWithOptions(stdout, log.Options{Prefix: "debug"}),
+		traceLogger:     log.NewWithOptions(stdout, log.Options{Prefix: "trace"}),
+		traceSubsystems: make(map[string]bool),
+	}
+}
+
+func (l *Logger) Printf(format string, args ...interface{}) {
+	l.infoLogger.Print(fmt.Sprintf(format, args...))
```

Comment:
nit: could have avoided to use fmt here:

```go
l.infoLogger.Printf(format, args...)
```

Reply:
fixed !